### PR TITLE
Feature/issue 624,751,809,803 -Remote Control Modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)

### LINKED ISSUE #624: [SDL 0099] New remote control modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)
Proposal: [New remote control modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data)](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0099-new-remote-control-modules-and-parameters.md)

> SDL remote control provides the ability for mobile applications to control certain components of the vehicle, such as, tune the radio, set the desired temperature or change the fan speed. However, currently only `CLIMATE` and `RADIO` modules are available for remote-control. Mobile application developers need more than basic climate or radio control. To meet developers' requirements, we propose to add three new modules (`AUDIO`, `LIGHT`, `HMI_SETTINGS`) and some new parameters (heatedXXXEnable and radio sisData) to the existing `CLIMATE` and `RADIO` modules respectively. This proposal only defines new data types and new parameters added to the existing data types. It does not add or change any remote control RPCs.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/296

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with revisions. The author will revise the proposal to fix a few typos, as well as include descriptions for the groups of lights - listing which single light elements they typically include. It was also noted that limiting the scope of proposals, especially those related to Remote Control, in the future will help to make for easier understanding and discussion among SDLC Members.
>Proposal has now been revised to include the descriptions and fix the typos mentioned.

### PR DESCRIPTION
**Pull request by mrapitis**
Fixes #624, #751, #809, #803 

This PR is **ready** for review.

## Risk

This PR makes **minor** API changes.

## Testing Plan

Added new unit tests and expanded existing ones

## Summary

Added RPC's, structs, enums as defined in the RC Modules (LIGHT, AUDIO, HMI_SETTINGS) and parameters (SIS Data), RC Lights – More Names and Status Values, Audio Source AM/FM/XM/DAB, Updating DOP value range for GPS notification proposals

## CLA

- [x] I have signed the CLA

### CODE REVIEW

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/ClimateControlCapabilitiesTests.java [hunk 1]
```diff
@@ -17,144 +17,165 @@
 import java.util.List;
 
 /**
- * This is a unit test class for the SmartDeviceLink library project class : 
+ * This is a unit test class for the SmartDeviceLink library project class :
  * {@link com.smartdevicelink.rpc.ClimateControlCapabilities}
  */
-public class ClimateControlCapabilitiesTests extends TestCase{
```
[the remark this answers is not in the document]
Comment (mrapitis) [reply sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/ClimateControlCapabilitiesTests.java]: We can update to a single tab.  The existing tabbing in many of the test files has been a bit off.

[the remark this answers is not in the document]
Comment (theRealBrett) [reply sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/datatypes/ClimateControlCapabilitiesTests.java]: Understood, Just trying to keep things as consistent as possible. Thanks

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/ClimateControlData.java [hunk 2]
```diff
@@ -3,104 +3,181 @@
 import com.smartdevicelink.proxy.RPCStruct;
 import com.smartdevicelink.proxy.rpc.enums.DefrostZone;
 import com.smartdevicelink.proxy.rpc.enums.VentilationMode;
+
 import java.util.Hashtable;
 
-public class ClimateControlData extends RPCStruct{
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/ClimateControlData.java]: For history, is it possible to only change the additions and not the whole class? It also makes it easier to review.

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlCapabilities.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
     public static final String KEY_STATE_AVAILABLE= "stateAvailable";
     public static final String KEY_SIGNAL_STRENGTH_AVAILABLE= "signalStrengthAvailable";
     public static final String KEY_SIGNAL_CHANGE_THRESHOLD_AVAILABLE= "signalChangeThresholdAvailable";
+	public static final String KEY_SIS_DATA_AVAILABLE= "sisDataAvailable";
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlCapabilities.java]: spacing

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlCapabilities.java [hunk 4]
```diff
@@ -245,4 +246,25 @@ public void setSignalChangeThresholdAvailable(Boolean signalChangeThresholdAvail
...
+	}
+
+	/**
+	 * Gets the sisDataAvailable portion of the RadioControlCapabilities class
+	 *
+	 * @return Boolean - Availability of the getting HD radio Station Information Service (SIS) data.
+	 * True: Available, False: Not Available, Not present: Not Available.
+	 */
+	public Boolean getSisDataAvailable() {
+        return getBoolean(KEY_SIS_DATA_AVAILABLE);
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlCapabilities.java]: spacing

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlData.java [hunk 5]
```diff
@@ -20,6 +21,7 @@
     public static final String KEY_SIGNAL_CHANGE_THRESHOLD= "signalChangeThreshold";
     public static final String KEY_RADIO_ENABLE= "radioEnable";
     public static final String KEY_STATE= "state";
+	public static final String KEY_SIS_DATA= "sisData";
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/RadioControlData.java]: spacing

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/StationIDNumber.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Hashtable;
+
+public class StationIDNumber extends RPCStruct {
+	public static final String KEY_COUNTRY_CODE = "countryCode";
+	public static final String KEY_FCC_FACILITY_ID = "fccFacilityId";
+
+	public StationIDNumber() {}
+
+	public StationIDNumber(Hashtable<String, Object> hash) {
+        super(hash);
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/StationIDNumber.java]: there are a few areas in this class where spacing is off

**Comment by mrapitis** [comment]
@theRealBrett PR has been updated, please re-review when time permits.  Thanks!

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GPSData.java [hunk 7]
```diff
@@ -181,30 +181,10 @@ public GPSData(Hashtable<String, Object> hash) {
     /**
      * Constructs a newly allocated GPSData object
      */
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/GPSData.java]: Removing this old constructor is a major change. Please keep it and deprecate it.

File: sdl_android/src/androidTest/assets/xml/MOBILE_API_4.5.0.xml [hunk 8]
```diff
@@ -1688,55 +1688,55 @@
         </param>
         <param name="latitudeDegrees" type="Float" minvalue="-90" maxvalue="90" mandatory="true">
         </param>
-        <param name="utcYear" type="Integer" minvalue="2010" maxvalue="2100" mandatory="true">
```
Comment (theRealBrett) [inline sdl_android/src/androidTest/assets/xml/MOBILE_API_4.5.0.xml]: This file should not have any changes to it, as this is not part of the 4.5 spec

Comment (mrapitis) [reply sdl_android/src/androidTest/assets/xml/MOBILE_API_4.5.0.xml]: Would it be preferred for us to create a MOBILE_API_4.7.0.xml file with the API changes, add it to the project and reference it from: https://github.com/smartdevicelink/sdl_android/blob/develop/sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/RPCConstructorsTests.java#L28
(leaving MOBILE_API_4.5.0.xml unchanged)

Comment (theRealBrett) [reply sdl_android/src/androidTest/assets/xml/MOBILE_API_4.5.0.xml]: I need to look into this and will get back to you soon

File: sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/enums/LightNameTests.java [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+		LightName enumFrontRightTurnLight = LightName.valueForString(example);
+		example = "REAR_LEFT_FOG_LIGHT";
+		LightName enumRearLeftFogLight = LightName.valueForString(example);
+		example = "REAR_RIGHT_FOG_LIGHT";
+		LightName enumRearRightFogLight = LightName.valueForString(example);
+		example = "REAR_LEFT_TAIL_LIGHT";
+		LightName enumRearLeftTailLight = LightName.valueForString(example);
+		example = "REAR_RIGHT_TAIL_LIGHT";
+		LightName enumRearRightTailLight = LightName.valueForString(example);
+		example = "REAR_LEFT_BREAK_LIGHT";
```
Comment (theRealBrett) [inline sdl_android/src/androidTest/java/com/smartdevicelink/test/rpc/enums/LightNameTests.java]: These now need to be updated per a proposal update to `REAR_LEFT_BRAKE_LIGHT` and `REAR_RIGHT_BRAKE_LIGHT`

File: sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/enums/LightName.java [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+	FRONT_RIGHT_FOG_LIGHT,
+	FRONT_LEFT_DAYTIME_RUNNING_LIGHT,
+	FRONT_RIGHT_DAYTIME_RUNNING_LIGHT,
+	FRONT_LEFT_TURN_LIGHT,
+	FRONT_RIGHT_TURN_LIGHT,
+	REAR_LEFT_FOG_LIGHT,
+	REAR_RIGHT_FOG_LIGHT,
+	REAR_LEFT_TAIL_LIGHT,
+	REAR_RIGHT_TAIL_LIGHT,
+	REAR_LEFT_BREAK_LIGHT,
```
Comment (theRealBrett) [inline sdl_android/src/main/java/com/smartdevicelink/proxy/rpc/enums/LightName.java]: These now need to be updated per a proposal update to REAR_LEFT_BRAKE_LIGHT and REAR_RIGHT_BRAKE_LIGHT

**Comment by mrapitis** [comment]
@theRealBrett enums have been updated to reflect the updated proposal.  Please re-review when time permits.  Thanks!

**Pull request merged**